plugins/optimization-detective/optimization.php

```diff
@@ -120,7 +120,10 @@ function od_can_optimize_response(): bool {
 		// users, additional elements will be present like the script from wp_customize_support_script() which will
 		// interfere with the XPath indices. Note that od_get_normalized_query_vars() is varied by is_user_logged_in()
 		// so membership sites and e-commerce sites will still be able to be optimized for their normal visitors.
-		current_user_can( 'customize' )
+		current_user_can( 'customize' ) ||
+		// Page caching plugins can only reliably be told to invalidate a cached page when a post is available to trigger
+		// the relevant actions on.
+		null !== od_get_cache_purge_post_id()
```

Comment:
This was wrong!!! It should have been:

```php
null === od_get_cache_purge_post_id()
```

See https://github.com/WordPress/performance/pull/1659